pkg/engine2/operational_eval/vertex_property.go

```diff
@@ -160,11 +160,16 @@ func (v *propertyVertex) evaluateConstraints(sol solution_context.SolutionContex
 		return fmt.Errorf("could not get current value for %s: %w", v.Ref, err)
 	}
 
-	if currentValue == nil && setConstraint.Operator == "" && v.Template != nil && v.Template.DefaultValue != nil {
+	ctx := solution_context.DynamicCtx(sol)
+	defaultVal, err := v.Template.GetDefaultValue(ctx, dynData)
```

Comment:
not currently, we could add a method for that, although it wont actually speed up the performance because the first check in the computation is just does the field == nil so it would essentially be doing the same thing